src/main/webapp/app/overview/course-competencies/course-competencies-details.component.ts

```diff
@@ -49,15 +59,27 @@ export class CourseCompetenciesDetailsComponent implements OnInit, OnDestroy {
                 if (this.competencyId && this.courseId) {
                     this.loadData();
                 }
+                this.course = this.courseStorageService.getCourse(this.courseId);
             });
         }
+
+        this.dashboardFeatureToggleActiveSubscription = this.featureToggleService.getFeatureToggleActive(FeatureToggle.StudentCourseAnalyticsDashboard).subscribe((active) => {
+            this.dashboardFeatureActive = active;
+        });
+    }
+
+    ngOnDestroy(): void {
+        this.dashboardFeatureToggleActiveSubscription?.unsubscribe();
+        this.paramsSubscription?.unsubscribe();
     }
 
     private loadData() {
         this.isLoading = true;
-        this.competencyService.findById(this.competencyId!, this.courseId!).subscribe({
-            next: (resp) => {
-                this.competency = resp.body!;
+        forkJoin([this.competencyService.findById(this.competencyId!, this.courseId!), this.competencyService.getJoL(this.courseId!, this.competencyId!)]).subscribe({
+            next: ([competencyResp, judgementOfLearningResp]) => {
+                this.competency = competencyResp.body!;
+                this.judgementOfLearning = judgementOfLearningResp.body ?? undefined;
```

Comment:
Could we not just remove the `?? undefined`?
```suggestion
                this.judgementOfLearning = judgementOfLearningResp.body;
```

Reply:
No